fineract-provider/src/main/java/org/apache/fineract/infrastructure/crypt/service/EncryptionKeyStoreServiceImpl.java

```diff
@@ -0,0 +1,104 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.fineract.infrastructure.crypt.service;
+
+import com.sun.jersey.spi.resource.Singleton;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import org.apache.fineract.infrastructure.configuration.domain.ConfigurationDomainService;
+import org.apache.fineract.infrastructure.core.service.DateUtils;
+import org.apache.fineract.infrastructure.core.service.ThreadLocalContextUtil;
+import org.apache.fineract.infrastructure.crypt.domain.EncryptionKeys;
+import org.apache.fineract.infrastructure.crypt.utils.RSAEncryptionUtils;
+import org.joda.time.Seconds;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.cache.annotation.CacheEvict;
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.cache.annotation.Caching;
+import org.springframework.stereotype.Service;
+
+/**
+ * @author manoj
+ */
+@Service
+@Singleton
+public class EncryptionKeyStoreServiceImpl  implements EncryptionKeyStoreService{
+    private Map<String, EncryptionKeys> keyMap =  new ConcurrentHashMap<>();
+
+
+    private final ConfigurationDomainService configurationDomainService;
+    private final RSAEncryptionUtils rsaEncryptionUtils;
+
+    @Autowired
+    public EncryptionKeyStoreServiceImpl(ConfigurationDomainService configurationDomainService, RSAEncryptionUtils rsaEncryptionUtils) {
+        this.configurationDomainService = configurationDomainService;
+        this.rsaEncryptionUtils = rsaEncryptionUtils;
+    }
+
+    @Caching(evict = {
+            @CacheEvict(value = "encryptionKeys", key = "T(org.apache.fineract.infrastructure.core.service.ThreadLocalContextUtil).getTenant().getTenantIdentifier().concat(#type)")})
+    public void storeKey(String type, EncryptionKeys key){
+        this.keyMap.put(getKeyType(type), key);
+    }
+
+    private EncryptionKeys retrieveValidKey(String type){
+
+        EncryptionKeys keys = getKeys(type);
+        if(keys == null){
+            return null;
```

Comment:
null are scary bad creatures, best avoided. Would it be clearer to throw some Exception here (include the type, so if it happens, you have context and know why)?

Reply:
This is trying to retrieve a key from the memory. When there is no key available here (that is, when null is returned) the wrapper which calls this method is supposed to create a new key and store it.  do you really think an Exception throw and catch needed in this context? I don't.